lib/api.js

```diff
@@ -249,6 +249,13 @@ async function handleBranchUpdate(context, eventName, event) {
   }
 }
 
+async function handleWorkflowDispatch(context, eventName, event) {
+  const { ref } = event;
+  const branch = ref.substr(11);
```

Comment:
Maybe this could be extracted into a common function? The function can then also replace the code from above:
```js
  const { ref } = event;
  if (!ref.startsWith("refs/heads/")) {
    logger.info("Push does not reference a branch:", ref);
    return;
  }
  const branch = ref.substr(11);
```
(and possibly use something like `"refs/heads/".length` instead of 11, to make it clearer)

Reply:
I can extract the branch name function, but I think each relevant event handler needs to check if the branch name is falsey, to bail out of the handler early.